packages/components/src/Simple.tsx

```diff
@@ -144,15 +150,19 @@ class Simple extends React.Component<SimpleProps, SimpleState> {
                 }
                 this.renderCanvas();
               },
-              fetchResolver
+              this.props.imageResolver ?? fetchResolver
             ));
         if (node.firstChild !== null) {
           node.replaceChild(renderedState, node.firstChild);
         } else {
           node.appendChild(renderedState);
         }
+        // propagate state update
+        if (this.props.onFrame) {
+          this.props.onFrame(this.penroseState);
+        }
```

Comment:
I'm not super familiar with how all our state management works here; what's the broad idea behind moving this out of `tick` and into `renderCanvas`?